src/web/components/table/KapTable.tsx

```diff
@@ -27,18 +27,42 @@ export const KapTable = <Data extends { id: string }>(props: KapTableProps<Data>
 
     const [orderBy, setOrderBy] = useState<string>(colDefDefaultSort?.id ?? colDefs[0].id); // Default to the first column
     const [order, setOrder] = useState<Order>(colDefDefaultSort?.sort ?? 'asc'); // Default to ascending order
-
     const handleRequestSort = (_event: React.MouseEvent<unknown>, property: string) => {
         const isAsc = orderBy === property && order === 'asc';
         setOrder(isAsc ? 'desc' : 'asc');
         setOrderBy(property);
     };
 
+    const [filterBy, setFilterBy] = useState<string | undefined>();
+    const [filterValue, setFilterValue] = useState<unknown>();
+    const onFilter = (property: string, value: unknown) => {
+        setFilterBy(property);
+        setFilterValue(value);
+    };
```

Comment:
If they are always set together, maybe they should just be one state object.
Also we might want to `useCallback` on this?

Reply:
I wrapped it in `useCallback` 👍  76b6eb7b79516e69d013003884ff5a5d897d1455

I prefer keeping them as two separate primitives as it simplifies where they are used. They are only updated in this `onFilter` function, so we are not calling them in pair other places.